packages/walletconnect-connector/src/index.ts

```diff
@@ -88,18 +89,36 @@ export class WalletConnectConnector extends AbstractConnector {
       this.emit(URI_AVAILABLE, this.walletConnectProvider.wc.uri)
     }
 
-    const account = await this.walletConnectProvider
-      .enable()
-      .then((accounts: string[]): string => accounts[0])
-      .catch((error: Error): void => {
-        // TODO ideally this would be a better check
-        if (error.message === 'User closed modal') {
-          throw new UserRejectedRequestError()
+    let account: string
+    account = await new Promise<string>((resolve, reject) => {
+      const userReject = () => {
+        // Erase the provider manually
+        this.walletConnectProvider = undefined
+        reject(new UserRejectedRequestError())
+      }
+
+      // Workaround to bubble up the error when user reject the connection
+      this.walletConnectProvider.wc.on('disconnect', (_: Error, event: IInternalEvent) => {
```

Comment:
On my end, I'm unable to detect 'disconnect' action.
Is this action dispatched when we click 'close' button on trustwallet ( to reject session ) after scanning QR code?

Reply:
Yes, the `disconnect` event emitted when user click close button to reject the session connection. I have tested and worked for me on these mobile wallet:
* Metamask
* Trust Wallet
* Argent

By the way I just follow the official [README](https://github.com/WalletConnect/walletconnect-monorepo/blob/1.3.4/packages/clients/client/README.md#initiate-connection) and not so sure whether it works for all mobile wallet. Actually the package `@walletconnect/web3-provider` this one depending on also [implements similar logic](https://github.com/WalletConnect/walletconnect-monorepo/blob/b2cfd26b5648cf0fb884b8115a288977701d5116/packages/providers/web3-provider/src/index.ts#L345-L351) for this situation but don't know why it doesn't work.